src/data/hash_queue.cc

```diff
@@ -130,17 +128,12 @@ HashQueue::remove(HashQueueNode::id_type id) {
     // The hash chunk was not found, so we need to wait until the hash
     // check finishes.
     if (!result) {
-      pthread_mutex_lock(&m_done_chunks_lock);
-      done_chunks_type::iterator done_itr;
+      auto lock = std::unique_lock(m_done_chunks_lock);
+      std::condition_variable      cv;
 
-      while ((done_itr = m_done_chunks.find(hash_chunk)) == m_done_chunks.end()) {
-        pthread_mutex_unlock(&m_done_chunks_lock);
-        usleep(100);
-        pthread_mutex_lock(&m_done_chunks_lock);
-      }
+      cv.wait(lock, [this, hash_chunk] { return m_done_chunks.find(hash_chunk) != m_done_chunks.end(); });
```

Comment:
You need to need to have `chunk_done()` call `cv.notify_all`.

Reply:
Fixed.